templates/CRM/Contribute/Form/Contribution.tpl

```diff
@@ -488,7 +488,9 @@
      });
 
      function showHideCancelInfo(obj) {
-       if (obj.find(":selected").text() == 'Refunded' || obj.find(":selected").text() == 'Cancelled') {
+       var selected_status_id  = obj.find(":selected").val();
```

Comment:
Are you absolutely sure about this? (including backward compatibility and such?)